Conch/lib/Conch/Controller/DeviceReport.pm

```diff
@@ -26,8 +26,18 @@ sub process ($c) {
 	my $raw_report = $c->req->body;
 
 	my $hw_product_name = $device_report->{product_name};
-	my $maybe_hw =
-		Conch::Model::HardwareProduct->lookup_by_name($hw_product_name);
+	my $maybe_hw;
+
+	if ( $device_report->{device_type}
+		&& $device_report->{device_type} eq "switch" )
```

Comment:
For imminent future-proofing this test, we might want to lop PDUs (`pdu`) in with this as well.